algorithms/maths/fibonacci_matrix.py

```diff
@@ -0,0 +1,42 @@
+
+#functions for matrix power and multiplication
+def matr_mult(x,y): #returns product of 2x2 matrices
+    return [[x[0][0]*y[0][0] + x[0][1]*y[1][0],     x[0][0]*y[0][1] + x[0][1]*y[1][1]],
+            [x[0][0]*y[1][0] + x[1][0]*y[1][1],     x[0][1]*y[1][0] + x[1][1]*y[1][1]]]
+
+def matr_pow(A,k): #returns A^k for integer k > 0 
+    if k==1:
+        return A
+    else:
+        X = matr_pow(A,k/2)
```

Comment:
```suggestion
        X = matr_pow(A, k // 2)
```
the code needs some formatting refactoring, and integer division here